step-capstone/src/components/Finalized.js

```diff
@@ -0,0 +1,132 @@
+import React from 'react';
+import {useEffect, useState, useRef} from 'react';
+import ReactDOM from 'react-dom';
+import TravelObject from './TravelObject';
+import TimeLine from './TimeLine';
+export default function Finalized(props) {
+    const hashMap = new Map();
+    let prevDiv = useRef([]);
+    const [displayDate, setDisplayDate] = useState(props.startDate);
+
+    const handleDisplayDateChange = (newDisplayDate) => {
+        setDisplayDate(newDisplayDate);
+    }
+
+    const sameDate = (timeA, timeB) => {
+      return (timeA.getDate() === timeB.getDate() && timeA.getMonth() === timeB.getMonth() && timeA.getFullYear() === timeB.getFullYear());
+    }
+    const pushItemToArrayInHashMap = (item, dateKey) => {
+        if (!hashMap.has(dateKey)) {
+            hashMap.set(dateKey, [item]);
+        }
+        else {
+            let array = hashMap.get(dateKey);
+            array.push(item);
+            hashMap.set(dateKey, array);
+        }
+    }
+    const separateDates = () => {
+        for(let i = 0; i < props.list.length; i++) {
+            let item = props.list[i];
+            let startDate = item.startDate;
+            let endDate = item.endDate;
+            pushItemToArrayInHashMap(item, startDate.toDateString());
+            if (!sameDate(startDate, endDate)) {
+                pushItemToArrayInHashMap(item, endDate.toDateString());
+            }
+        }
+    }
+    const travelObjectstartDateComparator = (travelObjectA, travelObjectB) => {
+        if (travelObjectA.startDate ===  travelObjectB.startDate) {
+            return 0;
+        }
+        else if (travelObjectA.startDate >  travelObjectB.startDate) {
+            return 1;
+        }
+        else   {
+            return -1;
+        }
+    }
+    const sortHashMap = () => {
+        const iterator = hashMap[Symbol.iterator]();
+        for (let KV of iterator) {
+            let V = KV[1];
+            V.sort(travelObjectstartDateComparator);
+        }
+    }
+    
+    useEffect(() => {
+        setDisplayDate(props.startDate);
+    }, [props.startDate]);
+    
+    const getAssociatedDiv = (startDate) => {
+        let output = startDate.getHours().toString();
+        if (startDate.getMinutes() < 30) {
+            return output + ":00";
+        }
+        else {
+            return output + ":30";
+        }
+    }
+    const getHeightPercentage = (startDate, endDate) => {
+        let dif = endDate.getHours()*60 + endDate.getMinutes() - startDate.getHours()*60 - startDate.getMinutes();
+        return dif*48.0/30-16;
```

Comment:
The values here (especially the 48) seem like a result of tuning. For such values, it can be wise to create a global variable rather than set value throughout your code. That way, if you ever need to change it, you can change it in one place, and it's clear what the value does up front.

Reply:
That's a good suggestion. I will make sure to put these const as a global variable.